blockdag/reachabilitystore.go

```diff
@@ -125,36 +125,55 @@ func (store *reachabilityStore) init(dbTx database.Tx) error {
 	//   with other nodes, which are now guaranteed to exist
 	cursor := bucket.Cursor()
 	for ok := cursor.First(); ok; ok = cursor.Next() {
-		err := store.loadReachabilityDataFromCursor(cursor, false)
+		err := store.initReachabilityData(cursor)
 		if err != nil {
 			return err
 		}
 	}
 	cursor = bucket.Cursor()
 	for ok := cursor.First(); ok; ok = cursor.Next() {
-		err := store.loadReachabilityDataFromCursor(cursor, true)
+		err := store.loadReachabilityDataFromCursor(cursor)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (store *reachabilityStore) loadReachabilityDataFromCursor(cursor database.Cursor, withConnections bool) error {
+func (store *reachabilityStore) initReachabilityData(cursor database.Cursor) error {
 	var hash daghash.Hash
 	err := hash.SetBytes(cursor.Key())
 	if err != nil {
 		return err
 	}
-	reachabilityData, err := store.deserializeReachabilityData(cursor.Value(), withConnections)
+
+	store.loaded[hash] = &reachabilityData{
+		treeNode:          &reachabilityTreeNode{},
+		futureCoveringSet: make(futureCoveringBlockSet, 0),
+	}
+	return nil
+}
+
+func (store *reachabilityStore) loadReachabilityDataFromCursor(cursor database.Cursor) error {
+	var hash daghash.Hash
+	err := hash.SetBytes(cursor.Key())
+	if err != nil {
+		return err
+	}
+
+	data, ok := store.reachabilityDataByHash(&hash)
```

Comment:
The name `data` is a bit problematic in this context, and even more so in `deserializeReachabilityData`'s context, since it does not hold any data at certain point.

I can't think of a better name RN, but try find something yourself, and if not, lets brainstorm together.